packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -514,135 +501,162 @@ const getAxesScale = (axis, scale, config) => {
-        config.allowCalibration,
         config.axis.y,
       );
 
-      const y2TickValues = generateYAxesTickValues(
-        config.axis.y2.domain.lowerLimit,
-        config.axis.y2.domain.upperLimit,
+      y2TickValues = generateYAxesTickValues(
+        y2LowerLimit,
+        y2UpperLimit,
         config.ticksCount,
-        config.allowCalibration,
         config.axis.y2,
       );
 
-      if (
-        config.axis.y.suppressTrailingZeros
-          && utils.isUndefined(config.axis.y.ticks.format)
-      ) {
-        axis.y = d3.axisLeft(scale.y);
-        tickFormatToTrimTrailingZeros = tickFormatter(axis.y);
-      }
-      axis.y = prepareYAxis(
-        scale.y,
-        yTickValues,
-        config.height,
-        getAxisTickFormat(
-          config.locale,
-          utils.isDefined(config.axis.y.ticks.format)
-            ? config.axis.y.ticks.format
-            : tickFormatToTrimTrailingZeros,
-        ),
-      );
-      tickFormatToTrimTrailingZeros = null;
-
-      if (
-        config.axis.y2.suppressTrailingZeros
-          && utils.isUndefined(config.axis.y2.ticks.format)
-      ) {
-        axis.y2 = d3.axisRight(scale.y2);
-        tickFormatToTrimTrailingZeros = tickFormatter(axis.y2);
-      }
-      axis.y2 = prepareY2Axis(
-        scale.y2,
-        y2TickValues,
-        config.height,
-        getAxisTickFormat(
-          config.locale,
-          utils.isDefined(config.axis.y2.ticks.format)
-            ? config.axis.y2.ticks.format
-            : tickFormatToTrimTrailingZeros,
-        ),
-      );
       // eslint-disable-next-line brace-style
-    }
+    } else {
     // Y and Y2 axes - If ticksCount is undefined or greater than
     // TICKS_MAXCOUNT AND if the Y2 is visible, then utilize a default value
     // for the ticksCount. This is based on the ranges of the Y & Y2 axes.
-    else {
-      const ticksCount = getAverageTicksCount(
-        config.axis.y.domain.upperLimit
-          - config.axis.y.domain.lowerLimit,
-        config.axis.y2.domain.upperLimit
-          - config.axis.y2.domain.lowerLimit,
-      );
 
-      const yTickValues = generateYAxesTickValues(
-        config.axis.y.domain.lowerLimit,
-        config.axis.y.domain.upperLimit,
-        ticksCount,
-        config.allowCalibration,
+      let intervalCount;
+
+      // Divide the larger range value (a) by the smaller value (b) and determine if a whole number (c) is returned.
+      // If the return value is a whole number and does not have a value grter than 10, divide the smaller range value (b) by the return value (c)
+      // and determine if a whole number is returned. If a whole number is returned, use the initial return value to configure the tick count of the graph.
+      const yRange = yUpperLimit - yLowerLimit;
+      const y2Range = y2UpperLimit - y2LowerLimit;
+      const greaterRange = yRange >= y2Range ? yRange : y2Range;
+      const lowerRange = yRange <= y2Range ? yRange : y2Range;
```

Comment:
You could use Math.min() and Math.max().

Reply:
Updated here: c70a5e131bc72a698715c2ff2ac0023705d22a46